src/shared/dropdown-list-v2/nested-dropdown-list.tsx

```diff
@@ -353,16 +370,24 @@ export const NestedDropdownList = <T,>({
         // Focus search input if there is one
         if (searchInputRef.current) {
             setFocusedIndex(-1);
+            setVirtuosoIndex(-1);
             setTimeout(() => searchInputRef.current?.focus(), 200); // wait for animation
         } else if (listItemRefs.current[focusedIndex]) {
             // Else focus on the specified element
             setTimeout(() => listItemRefs.current[focusedIndex]?.focus(), 200);
         } else {
             // Else focus on the first list item
             setFocusedIndex(0);
+            setVirtuosoIndex(0);
             setTimeout(() => listItemRefs.current[0]?.focus(), 200);
         }
-    }, [focusedIndex, mounted, setFocusedIndex]);
+    }, [
+        focusedIndex,
+        virtuosoIndex,
+        mounted,
+        setFocusedIndex,
+        setVirtuosoIndex,
```

Comment:
actually i think both of these setters can be removed as they come from useState and have a stable reference